internal/resources/pods.go

```diff
@@ -668,6 +658,13 @@ func defaultPodEnv(planner *sharePlanner) []corev1.EnvVar {
 	return env
 }
 
+func defaultPodSpec(planner *sharePlanner, sharens *bool) *corev1.PodSpec {
```

Comment:
Why is it returning a pointer here if all the other uses have to first dereference it? Wouldn't it be simpler to just return `corev1.PodSpec` here?

Reply:
You are right, will fix.

As a rule of thumb, I prefer to avoid aggregate return in C, and thus also in Golang (avoid redundant copies), although I am not sure if Go compiler will simply optimize the case above. In all odds, in the context of this patch it is just confusing. Will fix.